### PR TITLE
Migrate pkg/kubelet/prober to structured logging

### PR DESCRIPTION
**Pull request by krzysiekg**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Migrate pkg/kubelet/prober to structured logging

#### Which issue(s) this PR fixes:

Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/prober/prober_manager.go [hunk 1]
```diff
@@ -193,8 +192,8 @@ func (m *manager) AddPod(pod *v1.Pod) {
 		if c.LivenessProbe != nil {
 			key.probeType = liveness
 			if _, ok := m.workers[key]; ok {
-				klog.Errorf("Liveness probe already exists! %v - %v",
-					format.Pod(pod), c.Name)
+				klog.ErrorS(nil, "Liveness probe already exists for container",
```
Comment (yuzhiquan) [inline pkg/kubelet/prober/prober_manager.go:195]: That 's will be result in
```
I0102 15:04:05.067890    1234 prober_manager.go:956] "Liveness probe already exists for container"
```
beacuse err == nil will result in [this](https://github.com/kubernetes/klog/blob/110d443dcd6b29c2941cabd9ab9d211063658a21/klog.go#L809)

File: pkg/kubelet/prober/prober_manager.go [hunk 2]
```diff
@@ -181,8 +180,8 @@ func (m *manager) AddPod(pod *v1.Pod) {
 		if c.ReadinessProbe != nil {
 			key.probeType = readiness
 			if _, ok := m.workers[key]; ok {
-				klog.Errorf("Readiness probe already exists! %v - %v",
-					format.Pod(pod), c.Name)
+				klog.ErrorS(nil, "Readiness probe already exists for container",
```
Comment (yuzhiquan) [inline pkg/kubelet/prober/prober_manager.go:183]: As below.

File: pkg/kubelet/prober/prober_manager.go [hunk 3]
```diff
@@ -169,8 +168,8 @@ func (m *manager) AddPod(pod *v1.Pod) {
 		if c.StartupProbe != nil {
 			key.probeType = startup
 			if _, ok := m.workers[key]; ok {
-				klog.Errorf("Startup probe already exists! %v - %v",
-					format.Pod(pod), c.Name)
+				klog.ErrorS(nil, "Startup probe already exists for container",
```
Comment (yuzhiquan) [inline pkg/kubelet/prober/prober_manager.go:171]: As below.

File: pkg/kubelet/prober/prober_manager_test.go [hunk 4]
```diff
@@ -119,7 +118,7 @@ func TestAddRemovePods(t *testing.T) {
 
 	// Removing probed pod.
 	m.RemovePod(&probePod)
-	if err := waitForWorkerExit(m, probePaths); err != nil {
+	if err := waitForWorkerExit(t, m, probePaths); err != nil {
```
Comment (yuzhiquan) [inline pkg/kubelet/prober/prober_manager_test.go:121]: This change for trans klog to tesing log, may be in another pr.

File: pkg/kubelet/prober/prober_manager.go (hunk 1 above)
Comment (serathius) [reply pkg/kubelet/prober/prober_manager.go:195]: @yuzhiquan Problem you are mentioning should be fixed separately to migration. Let's continue with migration as described in migration guide.

**Comment by serathius** [comment]
Structured logging migration part looks good to me.

File: pkg/kubelet/prober/prober_manager_test.go (hunk 4 above)
Comment (krzysiekg) [reply pkg/kubelet/prober/prober_manager_test.go]: I removed changes to test files from this PR. I'll create a separate PR for that.

**Review by ehashman — COMMENTED** [review]
/lgtm

**Comment by krzysiekg** [comment]
@serathius I removed changes not related to structured logging migration from this PR. Could you give it a formal lgtm label?

File: pkg/kubelet/prober/prober_manager.go (hunk 1 above)
Comment (yuzhiquan) [reply pkg/kubelet/prober/prober_manager.go:195]: > @yuzhiquan Problem you are mentioning should be fixed separately to migration. Let's continue with migration as described in migration guide.

Sure.

**Comment by serathius** [comment]
@krzysiekg Yes, LGTM
/lgtm
